core/src/Kokkos_Layout.hpp

```diff
@@ -250,6 +251,15 @@ struct layout_iterate_type_selector<Kokkos::LayoutStride> {
   static const Kokkos::Iterate inner_iteration_pattern =
       Kokkos::Iterate::Default;
 };
+}  // namespace Impl
+
+// The layout_iterate_type_selector is now deprecated.
+// Please use the Kokkos::Impl::layout_iterate_type_selector instead.
```

Comment:
We should not suggest to use things in `Kokkos::Impl::`.  It will be obvious enough to developers and users are not allowed to reach for it.
```suggestion
```